### DISCUSSION TITLE
Statiq programs always returns 0, even when errors occurred

### DISCUSSION THREAD

**Discussion by girlpunk** [statiqdev/Discussions#191] [Ideas] [answered]
With the current example code, Statiq programs always return exit code 0. This means that CI/CD pipelines and other deployment methods are unable to automatically detect that something went wrong, and may proceed with trying to deploy the site anyway.

An easy solution to this is to change the example code to return the exit code that Statiq generates. For example, the current `Statiq.Docs` example code:

```csharp
await Bootstrapper
    .Factory
    .CreateDocs(args)
    .RunAsync();
```

is easily changed to

```csharp
return await Bootstrapper
    .Factory
    .CreateDocs(args)
    .RunAsync();
```

Is it worth changing the suggested code in the relevant places to make troubleshooting and automated deployment of Statiq sites easier?

**Comment by daveaglick** [comment, marked as answer]
You're absolutely right, all examples should be returning the exit code. Going through the documentation, I'm trying to find anywhere that it's not - I'm assuming you're talking about things like the partial second code sample here (as opposed to the first example, which does return the result code)?

![image](https://user-images.githubusercontent.com/1020407/217015425-6acde8da-21ac-424c-8f8e-90a7c3c6574a.png)

**Reply by daveaglick** [reply]
FYI - I went through and scrubbed the docs for situations like this and added `return` statements to all the ones I found, thanks a lot for the suggestion: https://github.com/statiqdev/Website/commit/543b1634c0d46df2f22963671cb30d39d5d6dda4

**Reply by girlpunk** [reply]
Yes, sorry I should have clarified which example I used. Thanks!
